demos/background_subtraction_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,213 @@
+        const float scale_y = static_cast<float>(model_in_size.height) / original_sz.height;
+        const float scale   = 1.f;
+
+        std::vector<cv::Mat> person_masks;
+        for (int box_idx = 0; box_idx < num_boxes; ++box_idx) {
+            const int conf_idx = box_idx * BOX_SIZE + CONFIDENCE_OFFSET;
+
+            if (boxes_ptr[conf_idx] > prob_threshold &&
+                labels_ptr[box_idx] == PERSON_ID) {
+                const int mask_offset     = box_idx * mask_sz.area();
+                const int box_pos_offset  = box_idx * BOX_SIZE;
+
+                auto box = expandBox(boxes_ptr[box_pos_offset + BOX_X0_OFFSET] / scale_x,
+                                     boxes_ptr[box_pos_offset + BOX_Y0_OFFSET] / scale_y,
+                                     boxes_ptr[box_pos_offset + BOX_X1_OFFSET] / scale_x,
+                                     boxes_ptr[box_pos_offset + BOX_Y1_OFFSET] / scale_y,
+                                     scale);
+
+                const auto& br  = box.br();
+                const auto& tl  = box.tl();
+
+                const int x0 = clip(tl.x, 0, original_sz.width);
+                const int y0 = clip(tl.y, 0, original_sz.height);
+                const int x1 = clip(br.x, 0, original_sz.width);
+                const int y1 = clip(br.y, 0, original_sz.height);
+
+                cv::Mat person_mask(mask_sz, CV_32FC1, masks_ptr + mask_offset);
+                cv::Mat resized_person_mask;
+
+                const auto diff = br - tl;
+                const cv::Size expanded_size(std::max(diff.x + 1, 1),
+                                             std::max(diff.y + 1, 1));
+                cv::resize(person_mask, resized_person_mask, expanded_size);
+
+                cv::Mat bin_mask = resized_person_mask > BIN_THRESHOLD;
+                bin_mask.convertTo(bin_mask, CV_8U, 255);
+
+                auto bin_mask_slice = cv::Rect(cv::Point(x0 - tl.x, y0 - tl.y),
+                                               cv::Point(x1 - tl.x, y1 - tl.y));
+
+                cv::Mat result_mask(original_sz, CV_8UC1, cv::Scalar(0));
+                bin_mask(bin_mask_slice).copyTo(result_mask(cv::Rect(cv::Point(x0, y0),
+                                                                     cv::Point(x1, y1))));
+                person_masks.push_back(result_mask);
+            }
+        }
+
+        if (person_masks.empty()) {
+            mask.setTo(255.f);
+        } else {
+            person_masks[0].copyTo(mask);
+            for (const auto& m : person_masks) {
+                mask |= m;
+            }
+        }
+    }
+};
+
+custom::NNBGReplacer::NNBGReplacer(const std::string& model_path) {
+    IE::Core core;
+    m_cnn_network = core.ReadNetwork(model_path);
+    m_tag         = m_cnn_network.getName();
+    m_inputs      = m_cnn_network.getInputsInfo();
+    m_outputs     = m_cnn_network.getOutputsInfo();
```

Comment:
```suggestion
    m_cnn_model = core.read_model(model_path);
    logBasicModelnfo(m_cnn_model);
    m_tag = m_cnn_model->get_friendly_name();
    m_inputs = m_cnn_model->inputs();
    m_outputs = m_cnn_model->outputs();
```
also new logging stage for demos was introduced recently, - [logBasicModelInfo](https://github.com/openvinotoolkit/open_model_zoo/blob/20dbbb69fcd7e4e0ea19f0386570fc2e72fba852/demos/common/cpp/utils/include/utils/common.hpp#L284).